### PR TITLE
Docker command for development

### LINKED ISSUE #138: Makefile for building B-Zone

### PR DESCRIPTION
**Pull request by Patricijia**
docker-compose -f docker-compose-dev.yml up

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ env:
 
 jobs:
   build-and-push:
-    runs-on: ubuntu-latest
```
Comment (NeaGogu) [inline .github/workflows/docker-image.yml]: no windows

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ on:
 jobs:
 
   build:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Comment (NeaGogu) [inline .github/workflows/go.yml]: no windows

File: BuildBackendDev.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+COPY backend/ .
+
+# Build the application inside the container
+RUN go build -o b-zone ./cmd/web
+
+ENV PORT=4000
+ENV NODE_ENV=development
+
+# Start the application
+CMD ["./b-zone"]
```
Comment (NeaGogu) [inline BuildBackendDev.Dockerfile]: this makes a very big image because you do everything in one step

File: BuildFrontendDev.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+WORKDIR /app
+
+# copy package.json and package-lock.json
+COPY package*.json ./
+
+# install dependencies
+RUN npm install
+
+# copy the rest of the application code
+COPY --from=builder /app/build .
```
Comment (NeaGogu) [inline BuildFrontendDev.Dockerfile]: you haven't defined any stage called builder

Comment (NeaGogu) [reply BuildFrontendDev.Dockerfile]: also you cannot use the --from tag unless you define another STEP (check the other Dockerfile for frontend)

File: BuildFrontendDev.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+# set environment variables
+ENV PORT=3000
+ENV NODE_ENV=development
```
Comment (NeaGogu) [inline BuildFrontendDev.Dockerfile]: how is node_env supposed to change behaviour ?

File: docker-compose-dev.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  b-zone-backend:
+    build:
+      #      context:
+      dockerfile: BuildBackendDev.Dockerfile
+    image: bzone/backend_image:1.0.0
+    container_name: b-zone-backend-container-dev
+    ports:
+      - "4000:4000"
+    networks:
+      - bzone_network
```
Comment (NeaGogu) [inline docker-compose-dev.yml]: make a different network so it doesnt conflict  if "prod" and "dev" run at the same time

File: docker-compose-dev.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+services:
+  b-zone-backend:
+    build:
+      #      context:
+      dockerfile: BuildBackendDev.Dockerfile
+    image: bzone/backend_image:1.0.0
```
Comment (NeaGogu) [inline docker-compose-dev.yml]: name the image differently if you want to have a different one for "dev'

File: docker-compose-dev.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    container_name: b-zone-backend-container-dev
+    ports:
+      - "4000:4000"
+    networks:
+      - bzone_network
+  b-zone-frontend:
+    build:
+         dockerfile: BuildFrontendDev.Dockerfile
+    image: goglete/sep_bzone_frontend:latest
```
Comment (NeaGogu) [inline docker-compose-dev.yml]: here you are specifying a build directive but you point to an image that's on my docker hub. Change the name of the image

**Review by NeaGogu — CHANGES_REQUESTED** [review]

**Comment by Patricijia** [comment]
@NeaGogu works for me with docker-compose -f docker-compose-dev.yml up -d, good job

**Review by PhilipWaldman — APPROVED** [review]

**Review by VictorVader — APPROVED** [review]

**Review by NeaGogu — APPROVED** [review]

**Pull request merged**